lib/web_ui/lib/src/engine/dom_renderer.dart

```diff
@@ -503,6 +503,74 @@ flt-glass-pane * {
     }
   }
 
+  static const String orientationLockTypeAny = 'any';
+  static const String orientationLockTypeNatural = 'natural';
+  static const String orientationLockTypeLandscape = 'landscape';
+  static const String orientationLockTypePortrait = 'portrait';
+  static const String orientationLockTypePortraitPrimary = 'portrait-primary';
+  static const String orientationLockTypePortraitSecondary = 'portrait-secondary';
+  static const String orientationLockTypeLandscapePrimary = 'landscape-primary';
+  static const String orientationLockTypeLandscapeSecondary = 'landscape-secondary';
+
+  /// Sets preferred screen orientation.
+  ///
+  ///
+  /// Specifies the set of orientations the application interface can be
+  /// displayed in.
+  ///
+  /// The [orientations] argument is a list of DeviceOrientation values.
+  /// The empty list uses Screen unlock api and causes the application to
+  /// defer to the operating system default.
+  ///
+  /// See w3c screen api: https://www.w3.org/TR/screen-orientation/
+  Future<bool> setPreferredOrientation(List<dynamic> orientations) {
+    final html.Screen screen = html.window.screen;
+    if (screen != null) {
+      final html.ScreenOrientation screenOrientation =
+          screen.orientation;
+      if (screenOrientation != null) {
+        if (orientations.isEmpty) {
+          screenOrientation.unlock();
+          return Future.value(true);
+        } else {
+          String lockType = _deviceOrientationToLockType(orientations.first);
+          if (lockType != null) {
+            final Completer<bool> completer = Completer<bool>();
+            try {
+              screenOrientation.lock(lockType).then((dynamic _) {
+                completer.complete(true);
+              }).catchError((dynamic error) {
+                // On Chrome desktop NotSupported on this device error is fired.
```

Comment:
nit: check that the caught error is indeed the Chrome NotSupported error?

Reply:
Updated comment. dynamic since it is W3C Draft.